app/views/browse/service-instance.html

```diff
@@ -145,7 +145,7 @@ <h3 ng-if-start="serviceClass.spec.description || serviceClass.spec.externalMeta
                           service-class="serviceClass"
                           service-plan="plan">
                         </service-instance-bindings>
-                        <div ng-if="(serviceInstance | serviceInstanceStatus) === 'Ready' && isMobileService">
+                        <div ng-if="(mobileClients | size) && (serviceInstance | serviceInstanceStatus) === 'Ready' && isMobileService">
```

Comment:
I want to get a wider opinion of this. Should we show the clients components only if there are clients?
Currently we show the Mobile Clients header and the Add Client button when no clients are in the namespace but with this change we don't show any header or button. There will be no mobile clients components visible.

Personally I prefer showing the header and button as it indicates at the possibly functionality. wdyt @maleck13 @witmicko @JameelB @vchepeli

Reply:
Does the header only show if mobile is enabled? I wouldn't want it to show even if mobile is not enabled. Also if we were to show the button without any client, should the add client button behaviour change to link to the catalog.

Reply:
yes only showing if mobile is enabled is what I would envision.

Since we can quickly search and provision apps from the search bar I'm not sure how advantageous linking to the catalog is. Maybe we can just have some text under the header? "Please provision a mobile client" or "No Mobile clients"

Reply:
I think we should follow what OpenShift do, When there is no resource, they don't show anything in the overview. The only exception to this is when there is nothing and they show the browse catalog button

Reply:
:+1: yep Im for following what Openshift do

Reply:
:+1: